quinn-proto/src/connection/streams.rs

```diff
@@ -490,6 +497,7 @@ impl Streams {
             // one.
             if let Ok(max) = VarInt::from_u64(self.local_max_data) {
```

Comment:
Should this be replaced with an `expect` now that we're being smart about setting `pending.max_data`?

Reply:
I think that would work in this place.

However there's still something not 100% correct:
- `local_max_data` can grow beyond `VarInt::MAX`, but we don't allow `sent_max_data` to grow earlier in. That means we might leave the last `receive_window / 8` bytes unused. I guess we could either address this, or just don't care about it - because nobody will ever send that much data.

Reply:
We could clamp to `VarInt::MAX` rather than omitting. On review, I don't think `expect` would be appropriate here anyway, since `local_max_data` might increase after `pending.max_data` is set, particularly with retransmits.

Reply:
Right - there exists a race! even without retransmits. Updated to use clamping